packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts

```diff
@@ -78,7 +83,19 @@ export class PayPalNotificationHandler extends PayPalHandler {
     }
   }
 
-  private async handleMpCancel(message: IpnMerchPmtType) {}
+  private async handleMpCancel(message: IpnMerchPmtType) {
+    const billingAgreement = await getPayPalBAByBAId(message.mp_id);
+    if (billingAgreement.status == 'Cancelled') {
+      return;
+    }
+    const accountCustomer = await getAccountCustomerByUid(billingAgreement.uid);
```

Comment:
What happens if we don't find an agreement in our system?

Reply:
That's a good point. I think we can just return and log that?

Reply:
Logging would be good, I could see this happening for example if we had already deleted a user before we get this event.